src/autogluon/bench/frameworks/multimodal/exec.py

```diff
@@ -17,57 +23,51 @@ def get_args():
     parser = argparse.ArgumentParser()
 
     parser.add_argument(
-        "--data_path",
+        "--dataset_name",
         type=str,
-        help="Can be one of: dataset name, local path, S3 path, AMLB task ID/name",
+        help="Dataset that has been registered with multimodal_dataset_registry.",
     )
+
     parser.add_argument("--benchmark_dir", type=str, help="Directory to save benchmarking run.")
+    parser.add_argument(
+        "--time_limit", type=int, default=None, help="Time limit for the AutoGluon benchmark (in seconds)."
+    )
+    parser.add_argument("--presets", type=str, default=None, help="Preset configurations to use in the benchmark.")
+    parser.add_argument("--hyperparameters", type=str, default=None, help="Hyperparameters to use in the benchmark.")
 
     args = parser.parse_args()
     return args
 
 
-def _convert_torchvision_dataset(dataset):
-    from io import BytesIO
-
-    import pandas as pd
-
-    data = []
-    for i in range(len(dataset)):
-        x, y = dataset[i]
-        img_byte_arr = BytesIO()
-        x.save(img_byte_arr, format="PNG")
-        img_byte_arr = img_byte_arr.getvalue()
-        data.append((img_byte_arr, y))
-    df = pd.DataFrame(data, columns=["image", "label"])
-    return df
-
-
 def load_dataset(
-    data_path: str,  # can be dataset name or path to dataset
+    dataset_name: str,  # dataset name
 ):
     """Loads and preprocesses a dataset.
 
     Args:
-        data_path (str): The path or name of the dataset to load.
+        dataset_name (str): The name of the dataset to load.
 
     Returns:
         Tuple[pd.DataFrame, pd.DataFrame]: A tuple containing the training and test datasets.
     """
-    if data_path in ["MNIST"]:
-        import torchvision.datasets as data
-
-        train_data = data.MNIST("./data", train=True, download=True)
-        test_data = data.MNIST("./data", train=False, download=True)
-        train_data = _convert_torchvision_dataset(train_data)
-        test_data = _convert_torchvision_dataset(test_data).sample(frac=0.1)
-    else:
-        df = load_pd.load(data_path)
-        train_data, test_data = train_test_split(df, test_size=0.2, random_state=42)
+    train_data = multimodal_dataset_registry.create(dataset_name, "train")
+    test_data = multimodal_dataset_registry.create(dataset_name, "test")
 
     return train_data, test_data
 
 
+class NumpyEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, np.integer):
+            return int(obj)
+        elif isinstance(obj, np.floating):
+            return float(obj)
+        elif isinstance(obj, np.ndarray):
+            return obj.tolist()
+        else:
+            return super(NumpyEncoder, self).default(obj)
+
+
```

Comment:
Should this go in a util file somewhere?

Reply:
Good point. Added an`utils.py` for `frameworks`.